test/test_cursor.py

```diff
@@ -665,27 +692,27 @@ def test_skip(self):
             self.assertEqual(i["x"], 0)
             break
 
-        for i in db.test.find().skip(20):
+        for i in (db.test.find()).skip(20):
```

Comment:
It looks like a large percentage of these changes are the extra parenthesis because async collection defined find() as async. We should consider making it a sync method since it's not intended to perform any I/O until it's first iterated anyway.

Reply:
`find` is only async because we need to call [_supports_exhaust](https://github.com/mongodb/mongo-python-driver/blob/d0193eb0456aebcd06615f4aa1647475bc79d57e/pymongo/asynchronous/cursor.py#L240) after creating a cursor. If we want to keep that existing behavior, we need to make `_server_property` and its use of `_get_topology()` and `select_server` synchronous.

Reply:
Instead of changing _server_property, AsyncCursor should delay checking exhaust support until the cursor is first iterated.